src/type_tracer/analyzer.rs

```diff
@@ -242,6 +243,7 @@ pub struct ModuleSymbol {
   swc_id_to_symbol_id: IndexMap<Id, SymbolId>,
   symbols: IndexMap<SymbolId, Symbol>,
   traced_re_exports: LockableRefCell<IndexMap<String, UniqueSymbolId>>,
+  traced_referrers: LockableRefCell<IndexMap<ModuleId, ImportedExports>>,
```

Comment:
I added this here: https://github.com/denoland/deno_graph/blob/ba14902441e250d8e384433d83466db259b46b0a/src/type_tracer/collections.rs#L67

It allows locking a RefCell so that a `&T` can be returned. This is useful here for a slightly better API that doesn't require borrowing, because once we've done the analysis pass, this will then become immutable. The LockableRefCell enforces this by panicking if anything calls `.borrow_mut()` once it's locked.

Reply:
That's a very handy API!